scalatest/src/main/scala/org/scalatest/RecoverMethods.scala

```diff
@@ -188,13 +188,18 @@ trait RecoverMethods {
   def recoverToSucceededIf[T <: AnyRef](future: Future[Any])(implicit classTag: ClassTag[T], exCtx: ExecutionContext, pos: source.Position): Future[Assertion] = {
     val clazz = classTag.runtimeClass
     future.failed.transform(
-      ex =>
+      rawEx => {
+        val ex =
+          rawEx match {
+            case execEx: java.util.concurrent.ExecutionException => execEx.getCause
+            case other => other
+          }
```

Comment:
Those exception that should be propogated thrown within the future are wrapped as ExecutionException, so here we extract them from ExecutionException and continue the checking.

Those failing tests are in the generated tests, to reproduce the problem, you can try remove the added code and run the following generated tests: 

rm -rf gentests
sbt genInspectorsTests/test

and see the result in the html reporter at target/html/index.html